src/server/capacity_unit_calculator.cpp

```diff
@@ -60,56 +60,90 @@ int64_t capacity_unit_calculator::add_write_cu(int64_t write_data_size)
     return write_cu;
 }
 
-void capacity_unit_calculator::add_get_cu(int32_t status, const dsn::blob &value)
+void capacity_unit_calculator::add_get_cu(int32_t status,
+                                          const dsn::blob &key,
+                                          const dsn::blob &value)
 {
-    if (status != rocksdb::Status::kOk && status != rocksdb::Status::kNotFound) {
+    if (status == rocksdb::Status::kNotFound) {
+        add_read_cu(kNotFound);
+        return;
+    }
+
+    if (status != rocksdb::Status::kOk) {
         return;
     }
-    add_read_cu(value.size());
+
+    add_read_cu(key.size() + value.size());
 }
 
 void capacity_unit_calculator::add_multi_get_cu(int32_t status,
+                                                const dsn::blob &hash_key,
                                                 const std::vector<::dsn::apps::key_value> &kvs)
 {
-    if (status != rocksdb::Status::kOk && status != rocksdb::Status::kNotFound &&
-        status != rocksdb::Status::kIncomplete && status != rocksdb::Status::kInvalidArgument) {
+    if (status == rocksdb::Status::kNotFound) {
+        add_read_cu(kNotFound);
+        return;
+    }
+
+    if (status != rocksdb::Status::kOk && status != rocksdb::Status::kIncomplete &&
+        status != rocksdb::Status::kInvalidArgument) {
         return;
     }
+
     int64_t data_size = 0;
     for (const auto &kv : kvs) {
         data_size += kv.key.size() + kv.value.size();
     }
+    data_size = data_size + hash_key.size();
     add_read_cu(data_size);
 }
 
 void capacity_unit_calculator::add_scan_cu(int32_t status,
                                            const std::vector<::dsn::apps::key_value> &kvs)
 {
-    if (status != rocksdb::Status::kOk && status != rocksdb::Status::kNotFound &&
-        status != rocksdb::Status::kIncomplete && status != rocksdb::Status::kInvalidArgument) {
+    if (status == rocksdb::Status::kNotFound) {
+        add_read_cu(kNotFound);
+        return;
+    }
+
+    if (status != rocksdb::Status::kOk && status != rocksdb::Status::kIncomplete &&
+        status != rocksdb::Status::kInvalidArgument) {
         return;
     }
+
     int64_t data_size = 0;
     for (const auto &kv : kvs) {
         data_size += kv.key.size() + kv.value.size();
     }
     add_read_cu(data_size);
 }
 
-void capacity_unit_calculator::add_sortkey_count_cu(int32_t status)
+void capacity_unit_calculator::add_sortkey_count_cu(int32_t status, const dsn::blob &hash_key)
 {
+    if (status == rocksdb::Status::kNotFound) {
+        add_read_cu(kNotFound);
+        return;
+    }
+
     if (status != rocksdb::Status::kOk) {
         return;
     }
-    add_read_cu(1);
+
+    add_read_cu(hash_key.size());
```

Comment:
`sortkey_count` is used to get sortkey count for a specific `hashkey`. Is it suitable to calculate cu only with hashkey size?

Reply:
Well, I have considered it when I refactor it, but `count` type operations have not reference from aliyun. I consider if it's appropriate that based `count` value?

Reply:
`sortkey_count` operation is similar to `hash_scan`, it should count key and value size during iterating rocksdb, the size should be `hashkey_size + all sortkey size + all value size`. 
However, if we calculate cu like above, we will add a lot codes in function `on_sortkey_count`. We can discuss it offline firstly.